### PR TITLE
Feature/global events tab

### PR DESCRIPTION
**Pull request by shanedeeodonnell**
@adaburrows @fosterlynn @zippy Simple event ledger. Would appreciate feedback on UI if you have any.

Screen Shot 2022-07-18 at 9 57 28 AM

### CODE REVIEW

**Comment by fosterlynn** [comment]
Nice!  Thoughts:

I'm wondering how it is sorted?  I also see discussion going on in hREA around ordering, so I'm not sure of any implications there.  So maybe it should stay however it is, and we can figure all that out later.

It would be helpful to have the process name included, if the event is input or output of a process. 

I think every event should show both the provider and receiver. (It might already do that, can't tell.)

Another note:  @adaburrows is working on the labels on the flows #72 , I'm wondering if you can make use of that logic when it is complete, since you are doing a bit of sentence structuring yourself?  Otherwise, maybe we should just do it more like a table, and if possible, do a table, don't know what the constraints are in the tech?  Also, I'm not advocating we do anything that will take too much time!

**Comment by shanedeeodonnell** [comment]
There isn't any sorting on it right now. Could add that in later, but likely not high-value right now.

Good idea, I'll add the process names, both in and out if applicable.

The events do show both provider and receiver, but if either is null then that one not displayed.

File: ui/src/data/models/Application/Display.ts [hunk 1]
```diff
@@ -117,3 +120,42 @@ export class DisplayEdge implements DisplayEdgeShape, PathedData {
     );
   }
 }
+
+export class DisplayEconomicEvent {
```
Comment (adaburrows) [inline ui/src/data/models/Application/Display.ts:124]: I'm confused about this. For one, there's already an implementation of `EconomicEvent` in `data/models/Valueflows/Observation.ts`:`EconomicEvent`.

The purpose of this file is act as a repository for the DisplayNode and DisplayEdge types that we store in a database. If there were any other objects that we store in the database and we display that have properties not in a Valueflows model, then we would put it here.

The `getNamesForId` seems like a helper that I've been putting in the `logic` directory. Continuing that tradition, the function should be renamed to `getAgentNamesForEconomicEvent`, rewritten to use the existing `EconomicEvent` model, and placed as a function in `ui/logic/flows.ts`.

I think that file is starting to get out of hand and we may need to start thinking about breaking it up into specific files used for specific views.

File: ui/src/components/layout/EventLedger.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    // loop through events and get names from IDs
+    for (const event of events) {
+      let displayEvent = new DisplayEconomicEvent(event);
+      await displayEvent.getNamesForId(event);
+      displayEvents.push(displayEvent);
+    }
+    setDisplayEconomicEvents(displayEvents);
+  }
+
+  const assembleCard = (econEvent: DisplayEconomicEvent) => {
```
Comment (adaburrows) [inline ui/src/components/layout/EventLedger.tsx:29]: This is typically the kind of thing I've been placing in the `ui/logic/flows.ts` file. I've basically been building a generic sort of library that builds strings and parts of views that is agnostic of React.

**Review by adaburrows — CHANGES_REQUESTED** [review]
Let's go over this in the morning.

**Comment by adaburrows** [comment]
Also, I am thinking it would be nice to have consistency between the view on the edges and this view. Perhaps we could make a component that takes a prop with a list of `EconomicEvents`.

**Comment by shanedeeodonnell** [comment]
rejecting because this feature has been better implemented in https://github.com/lightningrodlabs/rea-playspace/pull/102

**Pull request closed without merging**